src/client.rs

```diff
@@ -0,0 +1,850 @@
+use jsonrpc::error::Error;
+
+use super::primitives;
+
+pub struct Client {
+	agent: jsonrpc::client::Client
+}
+
+impl Client {
+	pub fn new(host: &str) -> Client {
+		Client { agent: jsonrpc::client::Client::new(host.to_owned(), None, None) }
+	}
+
+	pub fn new_with_credentials(host: &str, username: &str, password: &str) -> Client {
+		Client { agent: jsonrpc::client::Client::new(host.to_owned(), Some(username.to_owned()), Some(password.to_owned())) }
+	}
+
+	/// Returns a list of addresses owned by client.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// Vector of accounts owned by the client.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.accounts();
+	/// ```
+	pub fn accounts(&self) -> Result<Vec<primitives::Account>, Error>{
+		let request = self.agent.build_request("accounts", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<Vec<primitives::Account>>()) {
```

Comment:
The formatting here is a bit weird. Easiest way to fix is running `cargo fmt`. :)